### DISCUSSION TITLE
Locale masih error

### DISCUSSION THREAD

**Discussion by timomarwen** [teamalgoritma/community#64] [Q&A] [answered]
head(wday(vids$trending_date, #data yang akan di ekstrak informasinya
          label = T, #akan memberikan label nama hari : Sun, Mon, dkk
          abbr = F, #menampilkan nama panjang dari nama hari tsb. Eg: Sunday, Monday, etc.
          locale = "IND")) #untuk nama hari menggunakan local Indonesia. Eg : Senin, Selasa, etc

<div class="GEL-OVUBJGD ace_constant" style="display: inline; overflow-wrap: break-word; word-break: break-word; color: rgb(197, 6, 11); font-family: Monaco, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: pre-line; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-style: initial; text-decoration-color: initial;"><span>Error in Sys.setlocale("LC_TIME", locale) : 
(converted from warning) OS reports request to set locale to "IND" cannot be honored</span></div><span style="color: rgb(0, 0, 0); font-family: Monaco, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: pre-line; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(230, 230, 230); text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><div class="GEL-OVUBKGD stack_trace" aria-hidden="true" style="margin-top: 0.5em; margin-left: 0.3em; color: rgb(0, 0, 0); font-family: Monaco, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: pre-line; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-style: initial; text-decoration-color: initial; display: block;">

11. | doWithOneRestart(return(expr), restart)
-- | --

</div>

**Comment by onlyphantom** [comment, marked as answer]
First make sure you have Indonesian locale installed on your system. Open us a Terminal (bash or zsh in command prompt -- not Powershell or R console!) and run `locale -a`.

This is what is printed for mine (I'm on Ubuntu):
```
C
C.UTF-8
en_AG
en_AG.utf8
en_AU.utf8
en_BW.utf8
en_CA.utf8
en_DK.utf8
en_GB.utf8
en_HK.utf8
en_IE.utf8
en_IL
en_IL.utf8
en_IN
en_IN.utf8
en_NG
en_NG.utf8
en_NZ.utf8
en_PH.utf8
en_SG.utf8
en_US.utf8
en_ZA.utf8
en_ZM
en_ZM.utf8
en_ZW.utf8
id_ID.utf8
POSIX
```

I see that `id_ID.utf8` is what I want. If I try:
```
Sys.setlocale("LC_TIME", "IND")
```
I get an error:
> In Sys.setlocale("LC_TIME", "IND") :
  OS reports request to set locale to "IND" cannot be honored

If I try:
```
Sys.setlocale("LC_TIME", "id_ID.utf8")
```

It works successfully. So I will then modify the code to use that instead:

```
Sys.setlocale("LC_TIME", "id_ID.utf8")
weekdays(Sys.Date()+0:6)

# output:
[1] "Jumat"  "Sabtu"  "Minggu" "Senin"  "Selasa" "Rabu"   "Kamis" 
```

Depending on your OS (especially on non-Unix systems), you may not see `id_ID.utf8`. Change it accordingly and that will work! :)

If you want to reset, just change it back:

```
Sys.setlocale("LC_TIME", "en_GB.UTF-8")
weekdays(Sys.Date()+0:6)
# output:
[1] "Friday"    "Saturday"  "Sunday"    "Monday"    "Tuesday"   "Wednesday"
[7] "Thursday"
```
